tensorly/backend/core.py

```diff
@@ -738,7 +738,51 @@ def kr(self, matrices, weights=None, mask=None):
 
         return res*m
 
-    def partial_svd(self, matrix, n_eigenvecs=None, random_state=None, **kwargs):
+    def svd_flip(self, u, v, u_based_decision=True):
+        """Sign correction to ensure deterministic output from SVD.
+        Adjusts the columns of u and the rows of v such that the loadings in the
+        columns in u that are largest in absolute value are always positive.
+        This function is borrowed from scikit-learn/utils/extmath.py
+        Parameters
+        ----------
+        u : ndarray
+            u and v are the output of `partial_svd`
+        v : ndarray
+            u and v are the output of `partial_svd`
+        u_based_decision : boolean, (default=True)
+            If True, use the columns of u as the basis for sign flipping.
+            Otherwise, use the rows of v. The choice of which variable to base the
+            decision on is generally algorithm dependent.
+        Returns
+        -------
+        u_adjusted, v_adjusted : arrays with the same dimensions as the input.
+        """
+        if u_based_decision:
+            # columns of u, rows of v
+            is_numpy = isinstance(u, np.ndarray)
+            max_abs_cols = self.argmax(self.abs(u), axis=0)
+            signs = np.sign(self.to_numpy(u)[max_abs_cols, range(self.shape(u)[1])])
```

Comment:
I haven’t worked with Tensorflow, but it seems this kind of advanced indexing isn't supported. Is it possible to do this more elegantly for all backends?

Reply:
Thanks @merajhashemi! To do advanced indexing with all backends, you can use [tl.index_update](http://tensorly.org/dev/modules/generated/tensorly.index_update.html#tensorly.index_update).

Reply:
`tl.index_update` is helpful for assigning values to immutable tensors (JAX and TensorFlow) but doesn't support advanced indexing. 
Example:
```python
>>> x = tl.tensor([[1, 2], [3, 4], [5, 6]])
>>> # Supports all backends except tensorflow
>>> tl.index_update(x, tl.index[[0, 1, 2], [0, 1, 0]], 100)
array([[100,   2],
       [  3, 100],
       [100,   6]])
```
To support this kind of advanced indexing in TensorFlow we need to use`tf.gather_nd` like they do [here](https://github.com/jonasrauber/eagerpy/blob/master/eagerpy/tensor/tensorflow.py#L535-L577).

Reply:
We can add a `get_index` method to the backend:?

Reply:
Hmm I guess we could but it really adds another cumbersome methods just for tensorflow..
For this particular case, we could also just use:
```python
signs = tl.sign(tl.tensor([M[i, j] for (i, j) in zip(max_abs_cols, range(rank))], **tl.context(M)))
```